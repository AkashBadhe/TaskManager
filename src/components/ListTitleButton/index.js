import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { secondaryBackgroundColor, textColor } from "../../components/Theme";

const StyledListTitleButton = styled.button`
  flex-grow: 1;
  padding: 10px;
  font-weight: 600;
  display: flex;
  justify-content: space-between;
  background: ${secondaryBackgroundColor};
  color: ${textColor};
  font-size: 14px;
  border: none;
  cursor: pointer;
  border-top-left-radius: 3px;
  font-size: 1.2em;
`;

const ListTitleButton = ({ text, ...props }) => {
  return <StyledListTitleButton {...props}>{text}</StyledListTitleButton>;
};

ListTitleButton.propTypes = {
  text: PropTypes.string.isRequired,
};

export default ListTitleButton;
