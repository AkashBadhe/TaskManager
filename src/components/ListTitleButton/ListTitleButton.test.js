import React from "react";
import { shallow } from "enzyme";
import ListTitleButton from "./index";

describe("ListTitleButton", () => {
  let listTitleButton = shallow(<ListTitleButton text="Add" />);
  it("Should mount `ListTitleButton`", () => {
    expect(listTitleButton).toMatchSnapshot();
  });
});
