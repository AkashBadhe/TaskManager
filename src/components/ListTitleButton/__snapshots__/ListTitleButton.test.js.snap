// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTitleButton Should mount \`ListTitleButton\` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListTitleButton
    text="Add"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Add",
    },
    "ref": null,
    "rendered": "Add",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "baseHash": 400283751,
        "componentId": "sc-AxjAm",
        "isStatic": false,
        "rules": Array [
          "
  flex-grow: 1;
  padding: 10px;
  font-weight: 600;
  display: flex;
  justify-content: space-between;
  background: #f8f8f8;
  color: rgb(46, 68, 78);
  font-size: 14px;
  border: none;
  cursor: pointer;
  border-top-left-radius: 3px;
",
        ],
        "staticRulesId": "",
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-AxjAm",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Add",
      },
      "ref": null,
      "rendered": "Add",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "baseHash": 400283751,
          "componentId": "sc-AxjAm",
          "isStatic": false,
          "rules": Array [
            "
  flex-grow: 1;
  padding: 10px;
  font-weight: 600;
  display: flex;
  justify-content: space-between;
  background: #f8f8f8;
  color: rgb(46, 68, 78);
  font-size: 14px;
  border: none;
  cursor: pointer;
  border-top-left-radius: 3px;
",
          ],
          "staticRulesId": "",
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "sc-AxjAm",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
